### PR TITLE
DM-40426: Add new LATISS filters collimator and cyl_lens, and new gratings holo4_001 and pinhole mask options. 

### PR DESCRIPTION
**Pull request by edennihy**
Jenkins run -> https://rubin-ci.slac.stanford.edu/job/stack-os-matrix/36/

Note that the PR is organized in 3 separate commits, one for the new filters, one for the new grating, and the third is for the options of the pinhole masks. The pinhole masks (there are two) will only be installed in the grating wheel, but they have configurable options as far as the aperture size and aperture locations. The names were chosen to express all available configurations as it is unknown ahead of time which aperture size/location configuration will be used.

### CODE REVIEW

File: python/lsst/obs/lsst/filters.py [hunk 1]
```diff
@@ -291,10 +291,41 @@ def addFilter(filter_dict, band, physical_filter):
                      band="z"),
     FilterDefinition(physical_filter="SDSSy_65mm",
                      band="y"),
+    FilterDefinition(physical_filter="collimator",
+                     band="collimator"),
```
Comment (timj) [inline python/lsst/obs/lsst/filters.py:295]: We try not to name bands after filters if there is another option. We have a current list of: diffuser, u, g, r, i, z, y, unknown, notch, white.  Do collimator and cyl_lens have more general band names?

File: python/lsst/obs/lsst/filters.py [hunk 2]
```diff
@@ -291,10 +291,41 @@ def addFilter(filter_dict, band, physical_filter):
...
+                    "pinhole_2_2_0200",
+                    "pinhole_2_2_0100",
+                    "pinhole_2_3_1000",
+                    "pinhole_2_3_0500",
+                    "pinhole_2_3_0200",
+                    "pinhole_2_3_0100",
+                    "pinhole_2_4_1000",
+                    "pinhole_2_4_0500",
+                    "pinhole_2_4_0200",
+                    "pinhole_2_4_0100",
```
Comment (timj) [inline python/lsst/obs/lsst/filters.py:327]: Are the gratings turning up in files with these names or is there a requirement to derive the names from multiple FITS headers in the metadata translator?

**Review by timj — COMMENTED** [review]
I have a couple of comments before I do the review.

File: python/lsst/obs/lsst/filters.py (hunk 1 above)
Comment (edennihy) [reply python/lsst/obs/lsst/filters.py:295]: They do not have more general band names, but among that list I think either "white" or "unknown" would be the most appropriate as these two elements do not have any intentional attenuation by wavelength.

Comment (timj) [reply python/lsst/obs/lsst/filters.py:295]: Maybe use `white` then.

File: python/lsst/obs/lsst/filters.py (hunk 2 above)
Comment (edennihy) [reply python/lsst/obs/lsst/filters.py:327]: The gratings should be turning up in files with one of these names at a time. 

To expand a bit, the plan is to install the pinhole mask into the grating stage instrument with one of these names depending on the pinhole configuration we setup outside of the instrument, before the mask is installed. We chose this list of names to cover all possible configurations as we expect it to change over time. However, that change is always done as part of a grating swap.

File: python/lsst/obs/lsst/filters.py (hunk 1 above)
Comment (edennihy) [reply python/lsst/obs/lsst/filters.py]: Recommendation to use `white` adopted and pushed.

**Review by timj — APPROVED** [review]
Looks good.

**Pull request merged**